src/main/java/edu/harvard/iq/dataverse/DatasetFieldServiceBean.java

```diff
@@ -851,4 +852,73 @@ public String getFieldLanguage(String languages, String localeCode) {
         }
         return null;
     }
+
+    public List<DatasetFieldType> findAllDisplayedOnCreateInMetadataBlock(MetadataBlock metadataBlock) {
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<DatasetFieldType> criteriaQuery = criteriaBuilder.createQuery(DatasetFieldType.class);
+
+        Root<MetadataBlock> metadataBlockRoot = criteriaQuery.from(MetadataBlock.class);
+        Root<DatasetFieldType> datasetFieldTypeRoot = criteriaQuery.from(DatasetFieldType.class);
+
+        criteriaQuery.where(criteriaBuilder.and(
+                        criteriaBuilder.equal(metadataBlockRoot.get("id"), metadataBlock.getId()),
+                        datasetFieldTypeRoot.in(metadataBlockRoot.get("datasetFieldTypes")),
+                        criteriaBuilder.isTrue(datasetFieldTypeRoot.get("displayOnCreate"))
```

Comment:
Do we also need to say "or 'required' is True"? In my local DB producer name is required and displayoncreate is false - not sure it that's just because my local DB is messed up.